### DISCUSSION TITLE
Copilot failing to connect in VS Codium

### DISCUSSION THREAD

**Discussion by DjSapsan** [community/community#45377] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Hi. I just got Copilot and trying to use it in VS Codium. It is not available in the store so I downloaded and installed .VSIX file. Other extensions installed in this way are working perfectly fine, excep Copilot.
The error I have:
```
Invalid copilot token: missing token: 404
GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"
```
And a pop-up 
![image](https://user-images.githubusercontent.com/12209464/214694044-e6492851-1d13-4f5b-aa2a-597c4173b326.png)

Then I installed an official VS Code and Copilot is working perfectly fine. I suspect they will "not support the Codium", but I still hope for a help.

**Comment by tilmann1** [comment]
@DjSapsan That's correct, Copilot is not supported on VSCodium.

**Reply by kimja-muru-crasman** [reply]
What actions GitHub has planned on taking, to support Copilot on VSCodium? I use VS Codium over VS Code all the time, and 99% of the VS Code plugins I use do work perfectly on Codium, but not this one. I have 0 love for Microsoft due to it being a shady business with predatory practises, and I intend to avoid it as much as possible. (And yeah I know GitHub is owned by microsoft, duhh! Company decision to use GitHub, so that's unavoidable.) My point is: copilot works also with JetBrains, which to my knowledge is not a Microsoft product, so just do that, but with Codium? 

So what's the timeline on bringing Copilot to VS Codium?

**Comment by kalnode** [comment]
I've had the same experience in attempting to use Github Copilot in VS Codium. I'm switching to back to VS Code simply to be able to use Github Copilot. Most other major extensions work fine in VS Codium.

Since Microsoft _owns_ both Github and VS Code, haha, I think the chances are slim we'll see this extension work in VS Codium?

Anyways there's actually two concerns:

1. Primary: Make the extension work in VS Codium (problems seem to have to do auth and/or connectivity)
2. Secondary: Having the extension be registered on [Open VSX](https://open-vsx.org), so that it can appear when searching VS Codium's extensions panel (right now you have to manually download and install the VSX file from the MS marketplace)

**Comment by BackMountainDevil** [comment]
this is the info about it

```
[INFO] [auth] [2023-04-01T03:00:24.191Z] Invalid copilot token: missing token: 404 
[ERROR] [default] [2023-04-01T03:00:24.193Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"
```
- vscodium-bin: 1.77.0.23090-1
- github.copilot v1.78.9758

use iftop to know that auth api is `api.github.com`. I can ping that. I can sign in github succ. But fail with copilot. And I reinstall copilot and update vsc. So the point is copilot's trick?

**Comment by abcsec** [comment]
The bully who has won, but still doesn't realize it, for whatever reason, so keeps bullying everyone...? 

Besides their licensing conditions being monopolistic, it looks like the enforcement is in the extension too, so can't be circumvented with just a proxy.
